.github/workflows/gen-diffs.yaml

```diff
@@ -20,6 +20,10 @@ jobs:
 
     - name: Install sshpass, kicad, and tk bindings
       run: |
+        sudo cp /etc/apt/sources.list /etc/apt/sources.list.d/ubuntu-latest-mirrors.list
+        sudo sed 's#http://azure.archive.ubuntu.com/ubuntu/#http://ubuntu.osuosl.org/ubuntu/#' -i /etc/apt/sources.list.d/ubuntu-latest-mirrors.list
+        cat /etc/apt/sources.list | sudo tee -a /etc/apt/sources.list.d/ubuntu-latest-mirrors.list > /dev/null
+        sudo sed 's#http://azure.archive.ubuntu.com/ubuntu/#http://mirrors.ocf.berkeley.edu/ubuntu/#' -i /etc/apt/sources.list.d/ubuntu-latest-mirrors.list
         sudo add-apt-repository --yes ppa:kicad/kicad-6.0-releases
```

Comment:
Maybe this `add-apt-repository` command could be used instead of the copy/modify logic?  The copy/modify logic keeps all of the listed components and codename of the currently used mirror without having to specify them.